spec/lib/facilities/gis_client_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Facilities::GisClient do
+  RSpec::Matchers.define :has_order_and_offset do |order, offset|
+    match do |params|
+      params[:orderByFields].eql?(order) && params[:resultOffset].eql?(offset)
+    end
+  end
+
+  describe 'get_all_facilities' do
+    subject { described_class.new }
+
+    it 'passes the correct offset to the query when looping twice' do
+      r1 = double(Faraday::Response.new)
```

Comment:
I would rename these "r" variables to something more identifiable like `faraday_response_offset_0`. You can also declare each just once if you move them above of the describe block like `let(:faraday_response_offset_0)`.